### PR TITLE
[lm] 1) patch the luci/lucid files from within install.sh 2) update rpi firmware 3) update/fix ca-certificates Makefile

### PR DESCRIPTION
**Pull request by shmick**

### CODE REVIEW

**Comment by CapnBry** [comment]
Thanks, Steve, I'll take a look at this. I especially want to check to see why the luci patches aren't applying as they should (because you shouldn't have luci source code available to patch when install.sh is run)

**Comment by shmick** [comment]
Install.sh runs a feed update, at which point luci source is downloaded. After than the patching of the files is done.

Your old install.sh script used to put the files into feeds/luci/luci/patches, but there is no feeds/luci/luci directory after the git clone of luci. Creating a patches directory in feeds/luci and copying the files there also doesn't work as there doesn't appear to be anything in the luci Makefile to initiate patching of files.

**Comment by CapnBry** [comment]
Yeah I'll have to take a look though because the patching process is actually part of the main build process (built in the OpenWrt system, not the luci Makefile). If the git clone has a different directory layout then that could be the issue and that should be the way to go. The only time I use direct patching is when the build system won't apply patches itself.

File: openwrt/package/ca-certificates/Makefile [hunk 1]
```diff
@@ -7,11 +7,12 @@
...
 
 PKG_NAME:=ca-certificates
-PKG_VERSION:=20150426
+PKG_VERSION:=20151214
 
 PKG_SOURCE:=$(PKG_NAME)_$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=http://ftp.debian.org/debian/pool/main/c/ca-certificates
-PKG_MD5SUM:=717455f13fb31fd014a11a468ea3895d
+PKG_MD5SUM:=2233bfa64af6f58f5eca9735b6742818
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
```
Comment (shmick) [inline openwrt/package/ca-certificates/Makefile]: Won't compile without the PKG_BUILD_DIR line!

File: openwrt/package/ca-certificates/Makefile [hunk 2]
```diff
@@ -7,11 +7,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=ca-certificates
-PKG_VERSION:=20150426
+PKG_VERSION:=20151214
```
Comment (shmick) [inline openwrt/package/ca-certificates/Makefile]: Package version 20150426 doesn't exist anymore.

**Pull request closed without merging**